toolbox/bulk_processing/validators.py

```diff
@@ -141,10 +141,12 @@ def validate(value, **_kwargs):
 
 def region_matches_treatment_code():
     def validate(region, **kwargs):
-        if region[0] != kwargs['row']['TREATMENT_CODE'][-1]:
-            raise Invalid(
-                f'Region "{region}" does not match region in treatment code "{kwargs["row"]["TREATMENT_CODE"]}"')
-
+        try:
+            if region[0] != kwargs['row']['TREATMENT_CODE'][-1]:
+                raise Invalid(
+                    f'Region "{region}" does not match region in treatment code "{kwargs["row"]["TREATMENT_CODE"]}"')
+        except IndexError:
```

Comment:
This would also catch an index error on the treatment code, which may or may not be desirable... certainly, it's hard to understand the intent of the code, doing it this way. I'd have probably preferred to see it explicitly ignore blank regions... but that's a minor comment. The code _does_ fix the bug

Reply:
I have changed it to explicitly check for the length of region.

Reply:
Another option would be to use truthy evaluation like: `if region.strip() and region[0] != kwargs['row']['TREATMENT_CODE'][-1]:`. Happy to go for readability for this check tho 👍